docs/content/reference/name-collision.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Handling type naming collisions
+description: Examples of logic used to avoid type name collision
+linkTitle: Name Collision
+menu: { main: { parent: 'reference', weight: 10 }}
+---
+
+While most generated Golang types must have unique names by virtue of being based on their GraphQL `type` counterpart,
+which themselves must be unique, there are a few edge scenarios where conflicts can occur.  This document describes
+how those collisions are handled.
+
+## Enum Constants
+
+Enum type generation is a prime example of where naming collisions can occur, as we build the const names per value
+as a composite of the Enum name and each individual value.
+
+### Example Problem
+
+Currently, enum types are transposed as such:
+
+```graphql
+enum MyEnum {
+  value1
+  value2
+  value3
+  value4
+}
+```
+
+Which will result in the following Golang:
+
+```go
+type MyEnum string
+
+const (
+	MyEnumValue1 MyEnum = "value1"
+	MyEnumValue2 MyEnum = "value2"
+	MyEnumValue3 MyEnum = "value3"
+	MyEnumValue4 MyEnum = "value4"
+)
+```
+
+However, those above enum values are just strings.  What if you encounter a scenario where the following is
+necessary:
+
+```graphql
+enum MyEnum {
+  value1
+  value2
+  value3
+  value4
+	Value4
+	Value_4
+}
+```
+
+The `Value4` and `Value_4` enum values cannot be directly transposed into the same "pretty" naming convention as their
+resulting constant names would conflict with the name for `value4`, as so:
+
+```go
+type MyEnum string
+
+const (
+	MyEnumValue1 MyEnum = "value1"
+	MyEnumValue2 MyEnum = "value2"
+	MyEnumValue3 MyEnum = "value3"
+	MyEnumValue4 MyEnum = "value4"
+	MyEnumValue4 MyEnum = "Value4"
+	MyEnumValue4 MyEnum = "Value_4"
+)
+```
+
+Which immediately leads to compilation errors as we now have three constants with the same name, but different values.
+
+### Resolution
+
+1. Store each name generated as part of a run for later comparison
+2. Try pretty approach first.  Use if no conflicts
```

Comment:
Can you explain what the pretty approach is here? Thanks!

Reply:
Hmm, so the pretty approach is to use the raw output from the existing `ToGo` function.  I did not change that logic in this PR, and the logic in there is a bit tricky.  Would it be sufficient to say something like "Attempt to coerce name into CapitalCaseForm", or something?

Reply:
Yeah, that would be much more clear. Thanks!